docs/ref-manual/approx/array-hashes.md

```diff
@@ -0,0 +1,44 @@
+Array hashes
+============
+
+### Background: Non-Collision of Hashes is Essential for Storage Integrity
+
+Solidity makes intensive use of the Keccak hash function in order to create
+its storage layout.
+For instance consider this contract:
+```solidity
+contract A {
+  mapping(uint => uint) m; // first field of A -- gets slot 0
+
+  function setMTo0(uint i) {
+    m[i] = 0
+  }
+}
+
+```
+In order to determine the storage address of `m[i]`, solidity 
+computes this hash: `keccak(0, i)` (the first argument is 0 because mapping `m` 
+is the first field in contract `A`).
+
+This means that non-collision of hashes is crucial for memory integrity, since 
+a collision means that writes to different solidity variables interfere with 
+each other.
+
+### Modelling of Hashing in CVT -- Adequate, Except for Unbounded Arrays
+
+CVT generally uses a modelling of the Keccak function that prohibits these 
+unwanted collisions.
+An exception to this statement is the case when we hash arrays of unbounded 
+length, like `bytes`, or `string`.
```

Comment:
I might say "including `bytes` and `string`", since the most common case of unbounded arrays is something like `int[]`.